containers/autoReply/AutoReplyForm/useAutoReplyForm.js

```diff
@@ -0,0 +1,66 @@
+import { useState } from 'react';
+import moment from 'moment';
+
+const toModel = (AutoResponder) => {
+    const start = AutoResponder.StartTime * 1000;
+    const end = AutoResponder.EndTime * 1000;
+
+    const startDate = moment
+        .utc(start)
+        .startOf('day')
+        .valueOf();
+    const endDate = moment
+        .utc(end)
+        .startOf('day')
+        .valueOf();
+    const startTime =
+        moment
+            .utc(start)
+            .startOf('hour')
+            .add(30 * Math.floor(moment(start).minutes() / 30), 'minutes')
+            .valueOf() - startDate;
+    const endTime =
+        moment
+            .utc(end)
+            .startOf('hour')
+            .add(30 * Math.floor(moment(end).minutes() / 30), 'minutes')
+            .valueOf() - endDate;
+
+    return {
+        message: AutoResponder.Message,
+        duration: AutoResponder.Repeat,
+        daysOfWeek: AutoResponder.DaysSelected,
+        timeZone: AutoResponder.Zone,
+        subject: AutoResponder.Subject,
+        enabled: AutoResponder.IsEnabled,
+
+        startDate,
+        endDate,
+        startTime,
+        endTime
+    };
+};
+
+const toAutoResponder = (model) => ({
+    Message: model.message,
+    Repeat: model.duration,
+    DaysSelected: model.daysOfWeek,
+    Zone: model.timeZone,
+    Subject: model.subject,
+    IsEnabled: model.enabled,
+    StartTime: (model.startDate + model.startTime) / 1000,
+    EndTime: (model.endDate + model.endTime) / 1000
+});
+
+const useAutoReplyForm = (AutoResponder) => {
+    const [model, setModel] = useState(() => toModel(AutoResponder));
```

Comment:
Why the model is a `function`? Why about an `object`?

Reply:
My model is an object. When a function is passed to useState - it only calls it once on initial render. If I wrote `useState(toModel(AutoResponder));` my toModel function would get called every time the component re-rendered.

https://reactjs.org/docs/hooks-reference.html#lazy-initial-state